pkg/controller/ingress/concurrency_watchdog.go

```diff
@@ -336,3 +346,24 @@ func podIsReady(pod *corev1.Pod) bool {
 	}
 	return false
 }
+
+// podIsActive checks if a Pod is Ready and exists and is able to serve connections
+func podIsActive(ctx context.Context, lgr logr.Logger, cl client.Client, pod client.ObjectKey) (bool, error) {
+	var obj corev1.Pod
+	err := cl.Get(ctx, pod, &obj)
+	if apierrors.IsNotFound(err) {
+		lgr.Info("pod doesn't exist anymore")
+		return false, nil
+	}
+	if err != nil {
+		lgr.Error(err, "failed to get pod")
+		return false, fmt.Errorf("getting pod: %w", err)
```

Comment:
could you make it more specific than just "getting pod"? would help narrow down where the error is coming from

Reply:
that goes against Go best practices

https://google.github.io/styleguide/go/best-practices#adding-information-to-errors

It's on the caller to add information

Reply:
we log line number and function. I don't think it's a problem

Reply:
fair point - might be a good idea to add those columns to the ASI table since we usually look at logs there